src/renderer/inputSystem/armModeActions.ts

```diff
@@ -2,16 +2,25 @@ import { buttons as buttonMappings } from '@/renderer/inputSystem/mappings';
 import { rosClient } from '@/renderer/utils/ros/rosClient';
 import { Action } from '@/renderer/inputSystem/@types';
 import { TopicOptions } from '@/renderer/utils/ros/roslib-ts-client/@types';
-import { log } from '@/renderer/logger';
 import { ArmContext, armService } from '../state/arm';
 import { deadzone } from '../utils/gamepad';
 import { handleTpvControl } from './tpvControl';
+import { store } from '../store/store';
+import {
+  armPresetsSlice,
+  selectSelectedPreset,
+} from '../store/modules/armPresets';
 
 const jointGoalTopic: TopicOptions = {
   name: 'ovis/arm/in/joint_velocity_goal',
   messageType: 'ovis_msgs/OvisJointVelocity',
 };
 
+const jointPositionTopic: TopicOptions = {
+  name: 'ovis/arm/in/joint_position_goal',
+  messageType: 'ovis_msgs/OvisJointPosition',
+};
```

Comment:
You declare this twice exactly the same way. Here and in `src/renderer/hooks/useArmJointPositions.ts`.
Would there be a way to do it once to futureproof potential changes to the topic?

Reply:
The two are very similar but not exactly the same. They both use the same message type, but the topic name is different. One is a subscriber `ovis/arm/in/joint_position`, while the other is a publisher `ovis/arm/out/joint_position`. I was thinking of maybe creating a centralized topic files by modules containing all the topics so we can reuse them. But so far all the topics we use are only specific to one component.